src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -162,10 +163,21 @@ private Task RenderPageCoreAsync(IRazorPage page, ViewContext context)
                 foreach (var viewStart in viewStarts)
                 {
                     context.ExecutingFilePath = viewStart.Path;
+
                     // Copy the layout value from the previous view start (if any) to the current.
                     viewStart.Layout = layout;
+
                     await RenderPageCoreAsync(viewStart, context);
+
                     layout = viewStart.Layout;
+                    if (!string.IsNullOrEmpty(layout) && !IsApplicationRelativePath(layout) && IsRelativePath(layout))
+                    {
+                        // Fix up relative layout to be app-relative. Interpret layout relative to this view start.
+                        // Get directory name but do not use Path.GetDirectoryName() to preserve path normalization.
```

Comment:
Could we commonize this a `CombinePath` method? Pattern's identical to Line 228-231